control-plane/api-gateway/gatekeeper/poddisruptionbudget.go

```diff
@@ -0,0 +1,113 @@
+package gatekeeper
+
+import (
+	"context"
+	"github.com/hashicorp/consul-k8s/control-plane/api-gateway/common"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	v1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func (g *Gatekeeper) upsertPodDisruptionBudget(ctx context.Context, gateway gwv1beta1.Gateway, gcc v1alpha1.GatewayClassConfig, config common.HelmConfig) error {
+	if gcc.Spec.PodDisruptionBudgetSpec == nil || !gcc.Spec.PodDisruptionBudgetSpec.Enabled {
+		return g.deletePodDisruptionBudget(ctx, types.NamespacedName{Namespace: gateway.Namespace, Name: gateway.Name})
+	}
+
+	pdb := g.podDisruptionBudget(ctx, gateway, gcc, config)
+	mutated := pdb.DeepCopy()
+	mutator := newPodDisruptionBudgetMutator(pdb, mutated, gateway, g.Client.Scheme())
+
+	result, err := controllerutil.CreateOrUpdate(ctx, g.Client, mutated, mutator)
```

Comment:
Let's follow the pattern found in `role.go` where we check for an existing `PodDisruptionBudget` and, if one exists, verify that it's owned by our controller.

In the future, I'd like to use that pattern for all resources that we create, but that is outside the scope of this PR.

Reply:
my thought process was:

The configs remain same for resources that we create and the resources created through templates. i.e. they all have - enable, minAvailable and maxUnavailable thus making them homogenous.
For the resources we create, if the end-user/dev needs to disable PDB she/he could just go ahead and make enabled = false instead of deleting the whole stanza and adding it back again when they require it.
Shall I go ahead and remove the enabled flag anyways?